app/controllers/notes_controller.rb

```diff
@@ -318,24 +313,16 @@ def liked
 
   def recent
     @title = I18n.t('notes_controller.recent_research_notes')
-    @wikis = Node.limit(10)
-      .where(type: 'page', status: 1)
-      .order('nid DESC')
-    @notes = Node.where(type: 'note', status: 1, created: Time.now.to_i - 1.weeks.to_i..Time.now.to_i)
-                 .order('created DESC')
-    @unpaginated = true
+    @pagy, @notes = pagy(published_notes.where(created: Time.now.to_i - 1.weeks.to_i..Time.now.to_i)
+                 .order('created DESC'))
     render template: 'notes/index'
   end
 
   # notes with high # of views
   def popular
     @title = I18n.t('notes_controller.popular_research_notes')
-    @wikis = Node.limit(10)
-      .where(type: 'page', status: 1)
-      .order('nid DESC')
-    @notes = Node.research_notes
+    @notes = published_notes
```

Comment:
`.limit(20)` will take the first 20 records?
Do we need to sort the data in descending order of Node id (nid)? So we are sure we take latest 20 nodes and then find the popular ones?

Reply:
No limit(20) will not automatically take the first 20 records, I should sort the order then limit..you are right..let me change that..

Reply:
This has been fixed :)

Reply:
Awesome 🎉 
I had faced something similar before, so it immediately came to my notice 😄